# Cosine similarity

Compute cosine similarity between samples in X and Y.

Cosine similarity, or the cosine kernel, computes similarity as the normalized dot product of X and Y:

    K(X, Y) = <X, Y> / (||X||*||Y||)

* https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.cosine_similarity.html

Example:
Cosine similarity between [-2,2] and [1,0].

In [1]:
from sklearn.metrics.pairwise import cosine_similarity

print ( cosine_similarity([[-2,2], [1,0]]) )

[[ 1.         -0.70710678]
 [-0.70710678  1.        ]]


Cosine similarity is -0.70710678.

Example:
Cosine similarity between [0,0,0,1,1,1] and [1,1,1,0,0,0].

In [3]:
a = [0,0,0,1,1,1]
b = [1,1,1,0,0,0]

c = cosine_similarity([a,b])

print (c)

[[1. 0.]
 [0. 1.]]


Cosine similarity is 0.

In [4]:
a = [1,1,1]
b = [1,1,1]

c = cosine_similarity([a,b])

print (c)

[[1. 1.]
 [1. 1.]]


In [ ]:
Cosine similarity is 1.